convert2rhel/checks.py

```diff
@@ -715,106 +714,110 @@ def is_loaded_kernel_latest():
-            if (line.strip() and "listed more than once in the configuration" not in line.lower())
-        ]
-        # Filter out any error messages (things that do not have the three
-        # fields which we expect)
-        packages = [pkg for pkg in packages if len(pkg) == 3]
-
-        # Sort out for the most recent kernel with reverse order
-        # In case `repoquery` returns more than one kernel in the output
-        # We display the latest one to the user.
-        packages.sort(key=lambda x: x[0], reverse=True)
-
-        _, latest_kernel, repoid = packages[0]
-
-        # The loaded kernel version
-        uname_output, _ = run_subprocess(["uname", "-r"], print_output=False)
-        loaded_kernel = uname_output.rsplit(".", 1)[0]
-        match = compare_package_versions(latest_kernel, str(loaded_kernel))
-
-        if match == 0:
-            logger.info("The currently loaded kernel is at the latest version.")
-        else:
-            repos_message = (
-                "in the enabled system repositories"
-                if not reposdir
-                else "in repositories defined in the %s folder" % reposdir
-            )
-            logger.critical(
-                "The version of the loaded kernel is different from the latest version %s.\n"
-                " Latest kernel version available in %s: %s\n"
-                " Loaded kernel version: %s\n\n"
-                "To proceed with the conversion, update the kernel version by executing the following step:\n\n"
-                "1. yum install %s-%s -y\n"
-                "2. reboot" % (repos_message, repoid, latest_kernel, loaded_kernel, package_to_check, latest_kernel)
-            )
-    else:
-        # Repoquery failed to detected any kernel or kernel-core packages in it's repositories
-        # we allow the user to provide a environment variable to override the functionality and proceed
-        # with the conversion, otherwise, we just throw an critical logging to them.
-        unsupported_skip = os.environ.get("CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK", "0")
-        if unsupported_skip == "1":
-            logger.warning(
-                "Detected 'CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK' environment variable, we will skip "
-                "the %s comparison.\n"
-                "Beware, this could leave your system in a broken state. " % package_to_check
-            )
+    packages = []
+    # We are expecting an repoquery output to be similar to this:
+    #   C2R     1671212820      3.10.0-1160.81.1.el7    updates
+    # We need the `C2R` identifier to be present on the line so we can know for
+    # sure that the line we are working with is the a line that contains
+    # relevant repoquery information to our check, otherwise, we just log the
+    # information as debug and do nothing with it.
+    for line in repoquery_output.split("\n"):
+        if line.strip() and line.startswith("C2R"):
```

Comment:
```suggestion
        line = line.strip()
        if line.startswith("C2R"):
```

Okay my suggestion broke things because the line has initial whitespace.  This should fix my suggestion and I think it might be better too since:
* It sanitizes the data prior to using it which is the best way to organize things.
* The test for whether  line is not an empty string is a subset of the `line.startswith()` so we don't need to test them both independently.

Reply:
Tested it locally and it works! Thanks for fixing this for me, @abadger!